### PR TITLE
Make top level dependencies required

### PR DESCRIPTION
**Pull request by starksm64**
Add jakarta.enterprise.lang-model as a top level dependency

Signed-off-by: Scott M Stark <starksm64@gmail.com>

### CODE REVIEW

**Review by ivargrimstad — APPROVED** [review]

File: jakartaee-core-api/pom.xml [hunk 1]
```diff
@@ -75,42 +75,46 @@
...
             <version>${jakarta.enterprise.cdi-api.cp.version}</version>
-            <optional>true</optional>
+            <optional>false</optional>
             <exclusions>
-              <exclusion>
-                  <groupId>jakarta.inject</groupId>
-                  <artifactId>jakarta.inject-api</artifactId>
-              </exclusion>
+                <exclusion>
+                    <groupId>jakarta.inject</groupId>
```
Comment (Emily-Jiang) [inline jakartaee-core-api/pom.xml:111]: I am confused why you need to exclude these and then include them on the top level.

Comment (starksm64) [reply jakartaee-core-api/pom.xml:111]: It just makes it clear where the dependency is coming from. It is the way to be explicit about transitive dependencies.

**Comment by ivargrimstad** [comment]
I guess we will have to do this for Platform and Web Profile as well? 
In a separate PR, or bake it into this one?

**Comment by starksm64** [comment]
Yes, those need to be done as well. I can add it to this PR.

**Review by ivargrimstad — APPROVED** [review]

**Comment by starksm64** [comment]
API jars have been restaged

**Pull request merged**